### DISCUSSION TITLE
github server issue | just dont know

### DISCUSSION THREAD

**Discussion by Detroit-ish** [modelcontextprotocol/.github#125] [Q&A]
### Pre-submission Checklist

- [X] I have checked that this question would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions and documentation for answers

### Question Category

- [ ] Protocol Specification
- [ ] SDK Usage
- [X] Server Implementation
- [ ] General Implementation
- [ ] Documentation
- [ ] Other

### Your Question

Admittedly, long-time trier and lurcher. Yet, new-ish to code. But let's be honest—this is pretty straightforward.

So why am I hours into just these three, with both Git setups being the worst? The Git server—I’m not even sure how I got it working. Now the GitHub one? It connects, then disconnects.

I’ve created so many PATs, from full auth to minimal, both classic and fine-grained. I logged the env locally, ran it, and ran it through the terminal. Everything started fine. I close out Claude (a million times), and it just connects, then disconnects.

I can’t reason this out anymore. What am I missing?

Please and thanks,
Jason

{
  "mcpServers": {
    "weather": {
      "command": "node",
      "args": [
        "C:\\Users\\........"
      ]
    },
    "git": {
      "command": "uvx",
      "args": [
        "mcp-server-git",
        "--repository",
        "C:\\Users\\........"
      ]
    },
    "github": {
      "command": "npx",
      "args": [
        "-y",
        "@modelcontextprotocol\\server-github"
      ],
      "env": {
        "GITHUB_PERSONAL_ACCESS_TOKEN": "github_pat" 
      }
    }
  }
}

**Comment by Detroit-ish** [comment]
Again not a coder, so bare with me. 

To get it to work though, I uninstalled npm. Installed nvm, then installed and use node 22. It works now. 

Anyone can explain why, that would be cool!

**Reply by mgan59** [reply]
I can maybe help as I just set this up. By installing `nvm` it would of linked `npx` globally into your `env`, just run `which npx` from your terminal to verify.  When Claude Desktop Agent boots it can then use the same  _cli_ command to start the MCP agent server.  Why it felt like it magically just fixed itself.  The great thing about this approach is that the other nodejs/npm packages will work the same such as the __file-system__ mcp server.
